src/mapgen_functions.h

```diff
@@ -46,6 +46,12 @@ struct mapgendata
   bool is_groundcover(const ter_id iid ) const;
 };
 
+/**
+ * Calculates the coordinates of a rotated point.
+ * Should match the `mapgen_*` rotation.
+ */
```

Comment:
Would be helpful to state the center of the rotation. Is it (0,0)? Looks like it is (SEEX-1, SEEY-1). Very confusing. Also maybe mention what unit `turn` has (it's not in Degree, Radian nor Gradian). (Maybe also mention it does rotate in the XY-plane and therefor does not change `p.z`).

Anyway, nice to see some documentation.